src/middlewared/middlewared/plugins/iscsi_/alua.py

```diff
@@ -399,5 +406,132 @@ async def standby_after_start(self, job):
 
         job.set_progress(100, 'All targets in cluster_mode')
         self.standby_starting = False
-        self._all_cluster_mode = True
         self.logger.debug('ALUA started on STANDBY')
+
+    @job(lock='standby_reload', transient=True)
+    async def standby_reload(self, job):
```

Comment:
This is not a standby_reload function. If this gets called on the active controller, it will reload the service on this node.

Furthermore, the indiscriminate 30 second sleep causes me pause. Either way, the name of this method doesn't reflect reality unless I'm missing something obvious.

Reply:
This _only_ gets called when the `standby_node_requires_reload` flag gets set (because of a failure to communicate as expected with the other node).  And that flag only gets set in code protected by a check of `failover.status`.

That said it **certainly would** make sense to add a further check for `failover.status` BACKUP inside the function, immediately before we do the reload.

Wrt the 30 sec timer, well the two conditions we want to be able to recover from are:
1. Failure in `iscsi.target.login_ha_targets` (because the targets are not available yet from the ACTIVE node)
2. Failure in calling (`failover.call_remote`', '`iscsi.target.cluster_mode_targets_luns`')

Because this code is supposed to be a last-ditch fixup/recovery of an error path, I'm not sure if we can get more surgical in applying it.  Do you have a suggestion?